### PR TITLE
[MI-2441]: Added a skip button after an optional query

### PR DESCRIPTION
**Pull request by Nityanand13**

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -45,12 +45,54 @@ func (p *Plugin) initializeAPI() *mux.Router {
...
+		p.returnPostActionIntegrationResponse(w, response)
+		return
+	}
+
+	ctx := r.Context()
+	token := ctx.Value(ContextTokenKey).(*oauth2.Token)
+	userID := r.Header.Get(HeaderServiceNowUserID)
+
+	client := p.MakeClient(r.Context(), token)
+	if err := client.SendMessageToVirtualAgentAPI(userID, "_skip_internal", true, &MessageAttachment{}); err != nil {
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go]: Create a constant for this

File: server/plugin/virtual_agent.go [hunk 2]
```diff
@@ -488,8 +500,15 @@ func (p *Plugin) CreateTopicPickerControlAttachment(body *TopicPickerControl) *m
 	}
 }
 
-func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
+func (p *Plugin) CreateSkipButtonAttachment(message string) *model.SlackAttachment {
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go]:
```suggestion
func (p *Plugin) CreateOutputTextAttachment(message string) *model.SlackAttachment {
```

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/virtual_agent.go [hunk 3]
```diff
@@ -501,6 +520,21 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
 			},
 		},
 	}
+
+	if !body.Required {
+		slackAttachment.Actions = append(slackAttachment.Actions, p.PostActionToSkip())
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go:336]: Nice 👍🏽

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/constants.go [hunk 4]
```diff
@@ -23,6 +23,7 @@ const (
 	PathOpenDialog                 = "/api/v4/actions/dialogs/open"
 	PathSetDateTimeDialog          = "/date_time"
 	PathSetDateTime                = "/selected_date_time"
+	PathToSkip                     = "/skip"
```
Comment (Kshitij-Katiyar) [inline server/plugin/constants.go]: It should be similar to above paths as well like on line 23 variable name is PathOpenDialog not PathToOpenDialog.
```suggestion
	PathSkip                     = "/skip"
```

File: server/plugin/api.go [hunk 5]
```diff
@@ -45,12 +45,54 @@ func (p *Plugin) initializeAPI() *mux.Router {
...
+	decoder := json.NewDecoder(r.Body)
+	postActionIntegrationRequest := &model.PostActionIntegrationRequest{}
+	if err := decoder.Decode(&postActionIntegrationRequest); err != nil {
+		p.API.LogError("Error while decoding the PostActionIntegrationRequest params.", "Error", err.Error())
+		p.returnPostActionIntegrationResponse(w, response)
+		return
+	}
+
+	ctx := r.Context()
+	token := ctx.Value(ContextTokenKey).(*oauth2.Token)
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go]: Directly use this if possible.
```suggestion
	token := r.Context().Value(ContextTokenKey).(*oauth2.Token)
```

File: server/plugin/api.go [hunk 6]
```diff
@@ -45,12 +45,54 @@ func (p *Plugin) initializeAPI() *mux.Router {
...
+	userID := r.Header.Get(HeaderServiceNowUserID)
+
+	client := p.MakeClient(r.Context(), token)
+	if err := client.SendMessageToVirtualAgentAPI(userID, SkipInternal, true, &MessageAttachment{}); err != nil {
+		p.API.LogError("Error while sending the message to VA.", "Error", err.Error())
+		p.returnPostActionIntegrationResponse(w, response)
+		return
+	}
+
+	newAttachment := []*model.SlackAttachment{}
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go]: Directly initialize the array with the first element which you have appended below and fix this everywhere

File: server/plugin/virtual_agent_test.go [hunk 7]
```diff
@@ -306,6 +306,13 @@ func Test_CreatePickerAttachment(t *testing.T) {
 							},
 						},
 					},
+					{
+						Name: Skip,
```
Comment (Kshitij-Katiyar) [inline server/plugin/virtual_agent_test.go]: Above you have created a constant for Skip give a little more detailed name for the constant.

File: server/plugin/virtual_agent_test.go [hunk 8]
```diff
@@ -344,6 +351,13 @@ func Test_CreateDefaultDateAttachment(t *testing.T) {
 						},
 						Type: "button",
 					},
+					{
+						Name: Skip,
+						Type: "button",
```
Comment (Kshitij-Katiyar) [inline server/plugin/virtual_agent_test.go]: Check this everywhere
```suggestion
						Type: model.POST_ACTION_TYPE_BUTTON,
```

File: server/plugin/virtual_agent_test.go (hunk 7 above)
Comment (Nityanand13) [reply server/plugin/virtual_agent_test.go]: I have already used it

Comment (Kshitij-Katiyar) [reply server/plugin/virtual_agent_test.go]: better name for that constant

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/virtual_agent.go [hunk 9]
```diff
@@ -488,8 +500,15 @@ func (p *Plugin) CreateTopicPickerControlAttachment(body *TopicPickerControl) *m
 	}
 }
 
-func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
+func (p *Plugin) CreateOutputTextAttachment(message string) *model.SlackAttachment {
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go]:
```suggestion
func (p *Plugin) CreateOutputTextAttachmentWithSkipAction(message string) *model.SlackAttachment {
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**